### ISSUE TITLE
Support for ASP.NET Core 1.0 RC2

### ISSUE THREAD

**Issue by tsvetomir** [telerik/kendo-ui-core#1682]
We're currently working on upgrading our MVC Wrappers to the RC2 release. We'll post updates here.

**Comment by Eneuman** [comment]
Any ETA on this one ?

**Comment by tsvetomir** [comment]
There's no specific date yet, but we'll try to have it ready for the service release in the first half of June.

There are [45 breaking changes](https://github.com/aspnet/Announcements/issues?q=is%3Aopen+is%3Aissue+label%3A%22Breaking+change%22+milestone%3A1.0.0-rc2) for the RC2 release and this is just the _documented_ ones. So we've got our work cut out for us.

**Comment by tsvetomir** [comment]
Some fun facts to keep you entertained

* The MVC Core codebase is currently around 160 000 LOC of C#
* We started with ~ 3000 compilation errors after the RC2 upgrade
* Took us a day just to squash those out

**Comment by Eneuman** [comment]
Looks like you got your hands full ;)
Thanks for the update and keep up the good work!
Please drop a internal build once you got something for us to test. 
A lot of us are eager to upgrade our projects to rc2 :)
//Per

**Comment by Asshiah** [comment]
Yeah we are eager to try it out as soon as it is available even if it is just for testing purposes.^^

Thanks for all the hard work.

**Comment by kevinkoenig** [comment]
+1 for wanting it sooner rather than later.  Would be willing to beta test

**Comment by Vasimovic** [comment]
+1 sooner + beta

**Comment by nm-a** [comment]
+1 beta

**Comment by 1amirjalai** [comment]
sooner would be better we need to move to rc2 but we have decency on kendo ui

**Comment by Tiger333** [comment]
Yes would love to update to rc2 (some excellent new features added) but currently with a dependency on kendo ui so loooking forward to the (stable) release

Keep up the great work Telerik!

**Comment by nm-a** [comment]
any tentative date on beta, this will helpful to compile solution using RC2.

**Comment by tsvetomir** [comment]
We've got the majority of the functionality working with RC2. We're still catching up in areas like input validation that got a major overhaul in RC2.

We'll list any known issues, but things are lining up for a release in the beginning of next week.

**Comment by Tiger333** [comment]
Wow that would be awesome..fantsstic work

**Comment by ArieJones** [comment]
This would be awesome if we could get a release at the first of the week. We are in the middle of upgrading to RC2 and our last piece for this is the MVC wrapper stuff

**Comment by sophisma** [comment]
Is this still planed for a release in the next few days?

**Comment by tsvetomir** [comment]
The 2016.2.607 release is now live and supports the MVC Core 1.0 RC2 release.

Please, let us know if you find any issues. Cheers!

**Comment by marcels2204** [comment]
![dependencies](https://cloud.githubusercontent.com/assets/13157538/15868925/2a9829ec-2cea-11e6-95e2-a764cc0b2b09.png)

There are some dependency resolution errors. How to fix them?

**Comment by Akimi** [comment]
I have exactly same problem. Can't restore packages for `net452`. Can you add support for `net451` for `Kendo.Mvc`?

![err](https://cloud.githubusercontent.com/assets/8157374/15869061/f9c7e464-2cea-11e6-807e-d636829b1f07.png)

**Comment by thomasfreych** [comment]
@marcels2204 
Seems the packages are not available on nuget.org. The following steps solved the issue for me:

1. Add a new package source to https://www.myget.org/F/aspnetcidev/api/v3/index.json
2. Install packages (first select new Package source)
Install-Package Microsoft.Extensions.CompilationAbstractions -version 1.0.0-rc2-20221
Install-Package Microsoft.Dnx.Compilation.CSharp.Abstractions -Version 1.0.0-rc2-20221

Hope this helps.

**Comment by thomasfreych** [comment]
@Akimi 
You can change the target framework in the project.json:

  "frameworks": {
    "net452": { }
  },

Probably you get the same errors as @marcels2204 and have to take the same steps to solve it.

Hope this helps.

**Comment by Akimi** [comment]
@thomasfreych Installing missing packages from aspnetcidev solved my problem, thank you.

**Comment by marcels2204** [comment]
Why are the packages not available on nuget.org?

**Comment by thomasfreych** [comment]
@marcels2204 
Additional information about the branches and the package sources can be found here:
https://github.com/aspnet/Home/wiki/NuGet-feeds

**Comment by sophisma** [comment]
Hi guys, DNX was removed from RC2, maybe that DNX reference wasn't meant to be there.
I'm having the same issues for the following target frameworks:

**Comment by nm-a** [comment]
thank you @thomasfreych.

If these references can be removed, that would be great.
Microsoft.Extensions.CompilationAbstractions
Microsoft.Dnx.Compilation.CSharp.Abstractions

**Comment by alaatm** [comment]
How can we reference it from a library ?! library uses **netstandard1.5** then we get an incompatibility error. Shouldn't this be targeting netstandard1.x instead of netcoreapp1 ?

**Comment by kevinkoenig** [comment]
I am using Kendo Gird.  I get this error

      An unhandled exception has occurred while executing the request
System.NullReferenceException: Object reference not set to an instance of an object.
   at Kendo.Mvc.UI.WidgetBase.get_Id()
   at Kendo.Mvc.UI.Grid`1.ProcessSettings()
   at Kendo.Mvc.UI.WidgetBase.RenderHtml(TextWriter writer)
   at Kendo.Mvc.UI.WidgetBase.ToHtmlString()

for This View

    @(Html.Kendo().Grid<Unit>()
        .Name("UnitList")
        .DataSource(d => d
            .Ajax()
            .Read(r => r.Url("/Unit/Unit_Read").Type(HttpVerbs.Get))
            .Model(m=>m.Id(p=>p.Id)))
     )


I have removed most of the grid settings for simplicity.  Any ideas?

**Comment by underlog** [comment]
Hi guys, 

I would like to ask that you request assistance your specific problems (like the one from @kevinkoenig ) through our support system before this conversation branches too far. Thanks!

**Comment by kevinkoenig** [comment]
no but I will.  I did not know that was the preferred mechanism for support of rc bits.

**Comment by nm-a** [comment]
I am also getting same with Grid with RC2 fresh project

Code:
@(Html.Kendo().Grid<SendCodeViewModel>()
    .Name("UnitList")
    .Columns(columns =>
    {
        columns.Bound(p => p.ReturnUrl).Filterable(false);
        columns.Bound(p => p.RememberMe);        
    })
    .DataSource(d => d
        .Ajax()
        .Read(r =>  r.Action("Products_Read", "Home", null))
        )
)

Error:
NullReferenceException: Object reference not set to an instance of an object.
Kendo.Mvc.UI.Fluent.CrudOperationBuilderBase`1.Action(String actionName, String controllerName, RouteValueDictionary routeValues)
Asp._Views_Home_Contact_cshtml.<>c.<ExecuteAsync>b__17_2(CrudOperationBuilder r)
Kendo.Mvc.UI.Fluent.AjaxDataSourceBuilderBase`2.Read(Action`1 configurator)

**Comment by sophisma** [comment]
@underlog, aside from @kevinkoenig issue, all other replies are related to the 2016.2.607 release.
I believe there are still references to DNX and RC1 components in that release that removed on RC2.
You guys can check this by creating a new core web Project (RC2) and add the Kendo.Mvc nuget package to it. You'll see that the package won't be added because of those dependencies.

**Comment by Asshiah** [comment]
I agree with @sophisma.

This version is not fully migrated with RC2 and needs to be updated before we can use it to migrate our projects to RC2.

**Comment by nm-a** [comment]
hi @Asshiah 
I am also facing the same error with Grid. Here is the code snippet.

<script src="http://kendo.cdn.telerik.com/2016.2.607/js/kendo.all.min.js" type="text/javascript"></script>
<script src="http://kendo.cdn.telerik.com/2016.2.607/js/kendo.aspnetmvc.min.js" type="text/javascript"></script>

<link href="http://kendo.cdn.telerik.com/2016.2.607/styles/kendo.common.min.css" rel="stylesheet" type="text/css"/>

<div class="row">
@(Html.Kendo().Grid<SendCodeViewModel>()
    .Name("UnitList")
    .Columns(columns =>
    {
        columns.Bound(p => p.ReturnUrl).Filterable(false);
        columns.Bound(p => p.RememberMe);        
    })
    .DataSource(d => d
        .Ajax()
        .Read(r =>  r.Action("Products_Read", "Home", null))
        )
)

</div>

**Error:**
An unhandled exception occurred while processing the request.

NullReferenceException: Object reference not set to an instance of an object.
Kendo.Mvc.UI.Fluent.CrudOperationBuilderBase`1.Action(String actionName, String controllerName, RouteValueDictionary routeValues)
NullReferenceException: Object reference not set to an instance of an object.
Kendo.Mvc.UI.Fluent.CrudOperationBuilderBase`1.Action(String actionName, String controllerName, RouteValueDictionary routeValues)
Asp._Views_Home_Contact_cshtml.<>c.<ExecuteAsync>b__17_2(CrudOperationBuilder r)
Kendo.Mvc.UI.Fluent.AjaxDataSourceBuilderBase`2.Read(Action`1 configurator)

**Comment by nm-a** [comment]
is there any plan to update new version with fixes in this week?

**Comment by cjsmith26** [comment]
Is this a closed issue? I'm having the same problems as above.

**Comment by Asshiah** [comment]
I asked about this in the official forum in this post:
[http://www.telerik.com/forums/asp-net-core#PaQEnRSYG0mm29mFlCKlog](url)

**Comment by Tiger333** [comment]
Umm.. this is marked as closed but it appears from the comments that something is still not right here ?

**Comment by 1amirjalai** [comment]
new issue is created for this problem #1783

**Comment by Jekata** [comment]
I just answered this issue: [#1783](https://github.com/telerik/kendo-ui-core/issues/1783).

We do understand the inconvenience that the namespeace issue is causing and we are currently working with priority on it to solve it. Once this is done, we will release a new build.

**Comment by Jekata** [comment]
@marcels2204 @Akimi @1amirjalai and everyone else who wants to help: Would it be possible to provide a sample working project, which demonstrates the issue with the Microsoft.Extensions.CompilationAbstractions and Microsoft.Dnx.Compilation.CSharp.Abstractions namesapces?

We need such a project to test the hotfix before providing it publicly. 

Thank you in advance!

**Comment by Eneuman** [comment]
@Jekata 
This is how you reproduce the problem:
1) Remove any custom nuget feed you added to Visual Studio 2015. You should only have the default one listed.
2) Clear your nuget cache by "del %LOCALAPPDATA%\NuGet\Cache\*.nupkg /q"
3) Create a new ASP.Net Core Rc2 Web Project
4) Add Kendo MVC to dependencies in Project.json

Regards
Per

**Comment by Asshiah** [comment]
Here is a sample solution demonstrating the problem:

[DemoSolution](http://s000.tinyupload.com/?file_id=08788484944091094743)

**Comment by mparvanov** [comment]
The references problem described in [this post](https://github.com/telerik/kendo-ui-core/issues/1682#issuecomment-224364745) can be resolved by installing the referenced missing packages manually from:
[https://www.nuget.org/packages/Kendo.Mvc/](https://www.nuget.org/packages/Kendo.Mvc/).
In particular, these are:
[https://www.nuget.org/packages/Microsoft.Extensions.CompilationAbstractions/](https://www.nuget.org/packages/Microsoft.Extensions.CompilationAbstractions/)
[https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Formatters.Xml/](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Formatters.Xml/)
[https://www.nuget.org/packages/Microsoft.Dnx.Compilation.CSharp.Abstractions/](https://www.nuget.org/packages/Microsoft.Dnx.Compilation.CSharp.Abstractions/)

**Comment by 1amirjalai** [comment]
@mparvanov  these are dnx related packages. dnx was removed in asp.net core rc2

**Comment by Eneuman** [comment]
@mparvanov 
Yeah, but thats not a solution since you cannot do that stuff in you CI Build server or you will realy mess things up. Package 607 should be redacted since it's a faulty build.

**Comment by mparvanov** [comment]
That's absolutely right. We are currently researching this issue on our side with very high priority in order to provide a reliable resolution as soon as we could.

**Comment by tsvetomir** [comment]
Let's move this discussion to #1783
